src/orange/surf/SurfaceClipper.cc

```diff
@@ -0,0 +1,168 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/surf/SurfaceClipper.cc
+//---------------------------------------------------------------------------//
+#include "SurfaceClipper.hh"
+
+#include "orange/BoundingBoxUtils.hh"
+#include "orange/Constants.hh"
+
+#include "detail/AllSurfaces.hh"
+
+namespace celeritas
+{
+namespace
+{
+//---------------------------------------------------------------------------//
+#define ORANGE_INSTANTIATE_OP(IN)                                       \
+    template void SurfaceClipper::operator()(IN<Axis::x> const&) const; \
+    template void SurfaceClipper::operator()(IN<Axis::y> const&) const; \
+    template void SurfaceClipper::operator()(IN<Axis::z> const&) const
+
+constexpr real_type sqrt_half = constants::sqrt_two / 2;
+constexpr real_type sqrt_third = constants::sqrt_three / 2;
+//---------------------------------------------------------------------------//
+}  // namespace
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with interior and exterior bounding boxes.
+ */
+SurfaceClipper::SurfaceClipper(BBox* interior, BBox* exterior)
+    : int_{interior}, ext_{exterior}
+{
+    CELER_EXPECT(int_ && ext_);
+    CELER_EXPECT(encloses(*ext_, *int_));
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Clip the bounding boxes to an axis-aligned plane.
+ */
+template<Axis T>
+void SurfaceClipper::operator()(PlaneAligned<T> const& s) const
+{
+    ext_->shrink(Bound::hi, T, s.position());
+    int_->shrink(Bound::lo, T, s.position());
+}
+
+//!\cond
+ORANGE_INSTANTIATE_OP(PlaneAligned);
+//!\endcond
+
+//---------------------------------------------------------------------------//
+/*!
+ * Clip the bounding boxes to an axis-aligned cylinder.
+ */
+template<Axis T>
+void SurfaceClipper::operator()(CylCentered<T> const& s) const
+{
+    return (*this)(CylAligned<T>{s});
```

Comment:
Not in this case, no, but it can be a good way of enforcing that you don't discard the result of another function. Even for the void case it can say "I'm just going to pass this off to someone else".